stylix/mk-target.nix

```
@@ -0,0 +1,92 @@
+{
+  # name used for creation of options
+  name,
+
+  # name used for description of options
+  humanName,
+
+  # condition for automatically enabling this target
+  autoEnable ? true,
+
+  # options to be added under `options.stylix.targets.${name}`
+  extraOptions ? { },
+
+  # list of functions with args representing Stylix options; functions
+  # evaluated only if all args present as set attrs
```

Awesome, those docs really help clarify things for me as an outsider! Thanks

---

Is this wording nit slightly clearer?

```suggestion
  # are only evaluated if their 1st arg uses de-structured attrs syntax
```

---

Note 998d275218dce69440bf4e3ccf53d4f5715da090 still has the `{ name, value }` attrs needed for `listToAttrs`. These can go with `genAttrs`, since name is always the string value being mapped and value is always the thing returned by the mapping function.

https://noogle.dev/f/lib/genAttrs

---

I suppose that there are two things that need to be said here: first and most important is that if any of the requested Stylix attrs are set to `null`, that function's result does not pass a `lib.mkIf`; second is that the function arguments cannot have defaults, which is what I believe you are saying in your suggestion.

---

I have opted just to remove the argument default check as having defaults should make no difference in the current state of the function.